modules/rag.py

```diff
@@ -447,30 +446,29 @@ def eval_metrics(self, dataset_split, questions, predictions, references):
     
 
     def train(self):
-        from transformers import TrainingArguments
-        from transformers import AutoModelForCausalLM
-        from peft import LoraConfig, get_peft_model, prepare_model_for_kbit_training
-        from modules.trainer import RAGTrainer
         import torch
-        from transformers import Trainer
-        from modules.dataset import Tokenized_Sorted_Dataset
         from torch.utils.data import DataLoader
+        from transformers import TrainingArguments, Trainer
+        from peft import LoraConfig, get_peft_model, prepare_model_for_kbit_training
+        from modules.dataset import Tokenized_Sorted_Dataset
+        from modules.wandb_callback import WandbPredictionProgressCallback
```

Comment:
This import does not look necessary? Try,Catch around it to avoid dependencies to wandb?

Reply:
Solved